src/components/Button/Button.js

```diff
@@ -93,14 +95,37 @@ function PrivateButton({
     ''
   )
 
+  const [isDebounced, setIsDebounced] = useState(false)
+
+  useEffect(() => {
+    if (isDebounced) {
+      setTimeout(() => setIsDebounced(false), debounceDurationMs)
+    }
+  }, [isDebounced])
```

Comment:
May make sense to wrap this logic in a hook that can return the `debouncedOnClick` function

Reply:
updated, moved into custom hook https://github.com/getethos/ethos-design-system/pull/381/commits/dea7d6c8344728304db8362c2d0c94f69660b956